packages/curve-ui-kit/src/features/user-profile/UserProfileButton.tsx

```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react'
+import IconButton from '@mui/material/IconButton'
+import PersonIcon from '@mui/icons-material/Person'
+
+import { UserProfile } from './UserProfile'
+
+export const UserProfileButton = () => {
+  const [open, setOpen] = useState(false)
+
+  return (
+    <>
+      <IconButton size="small" onClick={() => setOpen(true)}>
+        <PersonIcon />
+      </IconButton>
+
+      <UserProfile open={open} onClose={() => setOpen(false)} />
```

Comment:
It's nice to lazy render the user profile as most times it will be invisible
```suggestion
      {open != null && <UserProfile open={open} onClose={close} />}
```

Reply:
`<Drawer open={open}` already lazily renders the user profile. Applying your suggestion also breaks the draw transition anyway.